### DISCUSSION TITLE
Quick Question About On-Demand Channels

### DISCUSSION THREAD

**Discussion by avalonMMXXII** [chrisbenincasa/tunarr#535] [Ideas]
I know the viewing habits have changed a bit...and DizqueTV addressed that by adding "On-Demand" live channels. Will this feature also be in Tunarr as well? 

Basically it runs the same a live channel, but pauses and resumes when the specific on-demand channel is viewed? But can still include watermarks, commercials, bumpers, intro videos, etc...

Is this feature going to also be migrated from DizqueTV to Tunarr?

**Comment by chrisbenincasa** [comment]
Hey sorry for the delay here. I can open an issue tracking this, but I'm not certain about its priority at the moment. To be honest, I never really understood the allure of on demand channels...I guess it's to have the features of DTV like filler but without a time schedule? Is that how you use it?

**Reply by Nouser** [reply]
My use case for On-Demand channels is that I don't have to leave the context of wherever I'm loading my DTV channels and I want to slowly work through serialized TV show(s) (where missing episodes or jumping around in seasons could render it unwatchable). While I could just watch the shows/channel in Plex, I have to keep track of watch status after looping through a show and it takes me out of where I'm watching my other shows.

I think of it as a playlist that saves my watch position independently of Plex/other channels (for media in multiple channels) and doesn't make me leave where I'm watching my channels. If I want to interleave episodes of 2 different series it becomes much easier using On-Demand Channels to watch them both serially without missing content.

**Reply by chrisbenincasa** [reply]
Thanks for the context - that makes sense.

Do you use the "segment length" feature with your on-demand channels? I've never used on-demand channels in DTV and I'm trying to figure out their behavior. The explanation of the field in the app doesn't really make much sense to me.

> Channel will be divided in segments. For example, if you use padding or time slots in your channel so that everything starts at 0:00 or 0:30 , you want a 30 minutes-segment. Use no segment if you want the channel to play exactly where you left it. Flex time will be added if necessary for padding.

My best guess now is that this just fast-forwards a channel to the nearest 'segment' upon resuming? Not really sure I'm a fan of that functionality, but curious what others think.

**Comment by chrisbenincasa** [comment]
Update here - I have a _mostly_ working implementation of on-demand channels in Tunarr. Right now, it does not support the "segmenting" functionality from DTV, simply because I don't really see the value in it? I mentioned in a reply in this discussion [here](https://github.com/chrisbenincasa/tunarr/discussions/535#discussioncomment-10034129) about that as well.

Need to do some touch-ups on the streaming changes (skipping Flex on resuming) and then get the UI hooked up, but it looks promising so far.

**Comment by chrisbenincasa** [comment]
Hey, just wanted to let you all know that I've merged in experimental support for on-demand channels with https://github.com/chrisbenincasa/tunarr/pull/613. Try it out (once it releases) and let me know what you think!
